custom_components/ramses_cc/binary_sensor.py

```diff
@@ -39,124 +80,214 @@ async def async_setup_platform(
     async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType = None,
 ) -> None:
-    """Create binary sensors for CH/DHW (heat) & HVAC.
-
-    discovery_info keys:
-      gateway: is the ramses_rf protocol stack (gateway/protocol/transport/serial)
-      devices: heat (e.g. CTL, OTB, BDR, TRV) or hvac (e.g. FAN, CO2, SWI)
-      domains: TCS, DHW and Zones
-    """
+    """Create binary sensors for CH/DHW (heat) & HVAC."""
 
     if discovery_info is None:
         return
 
     broker: RamsesBroker = hass.data[DOMAIN][BROKER]
 
-    def entity_factory(
-        broker: RamsesBroker, device, attr, *, entity_class=None, **kwargs
-    ):
-        return (entity_class or RamsesBinarySensor)(broker, device, attr, **kwargs)
-
-    new_sensors = [
-        entity_factory(broker, dev, k, **v)
-        for k, v in BINARY_SENSOR_ATTRS["gateway"].items()
-        if (dev := discovery_info.get("gateway"))
-    ]  # 18:xxxxxx - status
-    new_sensors += [
-        entity_factory(broker, dev, k, **v)
-        for key in ("devices", "domains")
-        for dev in discovery_info.get(key, [])
-        for k, v in BINARY_SENSOR_ATTRS[key].items()
-        if dev and hasattr(dev, k)
+    sensor_types: tuple[RamsesBinarySensorEntityDescription, ...] = (
+        RamsesBinarySensorEntityDescription(
+            key="gateway",
+            name="Gateway",
+            rf_class=Gateway,
+            entity_class=RamsesGatewayBinarySensor,
+            device_class=BinarySensorDeviceClass.PROBLEM,
+        ),
+        RamsesBinarySensorEntityDescription(
+            key="system",
```

Comment:
This sensor currently has an entity ID of `..._schema` but as it reflects the system availability `..._system` seemed like a better choice?

Reply:
Yes, I'd noticed this before - was on my (informal) to do list.

I am comfortable with 'minor' breaking changes, such as changing `entity_id`s for these non-core sensors.